restructure SSR options on page (#3378)
withastro/docs: src/content/docs/en/guides/server-side-rendering.mdx

## Enabling SSR in Your Project

> To enable SSR features in development mode, add one of the two `output` configuration options: `server` or (introduced in v2.6.0) `hybrid`. Both of these modes allow all or just some of your [pages](/en/core-concepts/astro-pages/) or [server endpoints](/en/core-concepts/endpoints/#server-endpoints-api-routes) to be server-rendered. Each configuration option has a different default behavior, and allows indidual routes to opt-out of the default accordingly:

Reviewer: ```suggestion
To enable SSR features for production deployments, update your `output` configuration to `'server'` or `'hybrid'` (introduced in **v2.6.0**). Both of these modes control which [pages](/en/core-concepts/astro-pages/) or [server endpoints](/en/core-concepts/endpoints/#server-endpoints-api-routes) should be server-rendered. Each configuration option has a different default behavior, and allows individual routes to opt-out of the default accordingly:
```

> - __`output: 'server'`__: Server-rendered by default. Any individual page or endpoint can be pre-rendered. Use this when most or all of your site should be server-rendered.

Reviewer: ```suggestion
- __`output: 'server'`__: Server-rendered by default. Use this when most or all of your site should be server-rendered. Any individual page or endpoint can *opt-in* to pre-rendering.
```

> - __`output: 'hybrid'`__: Pre-rendered to HTML by default. Any individual page or endpoint can *opt-out* of pre-rendering. Use this when most of your site should be static.

Reviewer: ```suggestion
- __`output: 'hybrid'`__: Pre-rendered to HTML by default. Use this when most of your site should be static. Any individual page or endpoint can *opt-out* of pre-rendering.
```
